src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/TokenizerBackedParser.cs

```diff
@@ -705,11 +705,15 @@ protected void ConfigureSpanContext(SpanContextConfigAction? config)
 
     protected void ConfigureSpanContext(SpanContextConfigActionWithPreviousConfig? config)
     {
-        var prev = SpanContextConfig;
         SpanContextConfig = config == null
             ? null
-            : ((SpanEditHandlerBuilder? span, ref ISpanChunkGenerator? chunkGenerator) => config(span, ref chunkGenerator, prev));
+            : GetNewSpanContextConfigAction(config, SpanContextConfig);
         InitializeContext();
+
+        static SpanContextConfigAction GetNewSpanContextConfigAction(SpanContextConfigActionWithPreviousConfig config, SpanContextConfigAction? prev)
```

Comment:
This is uncommon enough that it I think it probably warrants a comment that it needed to avoid allocations.